imports/plugins/core/cart/server/no-meteor/mutations/convertAnonymousCartToNewAccountCart.js

```diff
@@ -53,7 +56,12 @@ export default async function convertAnonymousCartToNewAccountCart({
   });
 
   const { deletedCount } = await Cart.deleteOne(anonymousCartSelector);
-  if (deletedCount === 0) throw new ReactionError("server-error", "Unable to delete anonymous cart");
+  if (deletedCount === 0) {
+    if (referenceId) {
+      await Cart.findOneAndUpdate({ _id }, {set: { referenceId } });
```

Comment:
Since you are not even using the `findOne` result, these should both be `Cart.updateOne` instead of `Cart.findOneAndUpdate`.